src/main/java/seedu/address/logic/commands/LessonEditCommand.java

```diff
@@ -0,0 +1,290 @@
+        return new CommandResult(String.format(MESSAGE_EDIT_LESSON_SUCCESS, toEdit,
+                editedLesson, personAfterLessonEdit), personAfterLessonEdit);
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code editPersonDescriptor}.
+     */
+    private static Person createEditedPerson(Person personToEdit, Lesson toEdit, Lesson editedLesson) {
+        assert personToEdit != null;
+
+        Set<Lesson> updatedLessons = new TreeSet<>(personToEdit.getLessons().stream()
+                .map(lesson -> lesson.equals(toEdit) ? editedLesson : lesson)
+                .collect(Collectors.toSet()));
+
+        return PersonUtil.createdEditedPerson(personToEdit, updatedLessons);
+    }
+
+    /**
+     * Creates and returns a {@code Lesson} with the details of {@code lessonToEdit}
+     * edited with {@code editPersonDescriptor}.
+     */
+    private static Lesson createEditedLesson(Lesson lessonToEdit, EditLessonDescriptor editLessonDescriptor) {
+        assert lessonToEdit != null;
+
+        Date updatedDate = editLessonDescriptor.getDate()
+                .orElse(lessonToEdit.getStartDate());
+        TimeRange updatedTimeRange = editLessonDescriptor.getTimeRange()
+                .orElse(lessonToEdit.getTimeRange());
+        Subject updatedSubject = editLessonDescriptor.getSubject()
+                .orElse(lessonToEdit.getSubject());
+        Set<Homework> updatedHomeworkSet = editLessonDescriptor.getHomeworkSet()
+                .orElse(lessonToEdit.getHomework());
+
+        return lessonToEdit.isRecurring()
+                ? new RecurringLesson(updatedDate, updatedTimeRange, updatedSubject, updatedHomeworkSet)
+                : new MakeUpLesson(updatedDate, updatedTimeRange, updatedSubject, updatedHomeworkSet);
+    }
+
+    @Override
+    protected void undo() {
+        requireNonNull(model);
+
+        model.setPerson(personAfterLessonEdit, personBeforeLessonEdit);
+    }
+
+    @Override
+    protected void redo() {
+        requireNonNull(model);
+
+        try {
+            executeUndoableCommand();
+        } catch (CommandException ce) {
+            throw new AssertionError(MESSAGE_REDO_FAILURE);
+        }
+
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof LessonEditCommand)) {
+            return false;
+        }
+
+        // state check
+        LessonEditCommand e = (LessonEditCommand) other;
+        return index.equals(e.index)
+            && indexToEdit.equals(indexToEdit);
+    }
+
+    /**
+     * Stores the details to edit the person with. Each non-empty field value will replace the
+     * corresponding field value of the person.
+     */
+    public static class EditLessonDescriptor {
+
+        private Date date;
```

Comment:
Not allowed to be edited (Remove from prefix check in LessonEditCommandParser). I left the fields for {@code date} and {@code isRecurring} in EditLessonDescriptor because it felt weird to not include them but the code doesn't need to use these fields... should I remove them

Reply:
Actually yeah, I will remove them